rakan/cRakan/src/Runner.h

```diff
@@ -134,74 +162,100 @@ class Runner {
+  * @param    beta          the weight to assign to the population distribution
+  *                         score
+  * @param    gamma         the weight to assign to the existing border
+  *                         respectfulness score
+  * @param    eta           the weight to assign to the VRA score
   * 
   * @return a sum of all the scores accumulated during the walk
   */
-  double Walk(uint32_t num_steps, double alpha, double beta, double gamma, double eta);
-
+  double Walk(uint32_t num_steps,
+              double alpha, double beta, double gamma, double eta);
 
 
  //////////////////////////////////////////////////////////////////////////////
  // Queries
  //////////////////////////////////////////////////////////////////////////////
 
-  /*
+  /**
   * Queries whether or not a district is empty.
   * 
-  * @param  district   The district to query
+  * @param  district   the district to query
   * 
   * @return true iff the district is empty
   */
   bool IsEmptyDistrict(int district);
 
-  /*
+  /**
   * Queries whether or not the district that the proposed node is in will be
   * severed once the proposed node is removed.
   * 
-  * @param    proposed_node   The node that will be hypothetically removed
+  * @param    proposed_node   the node that will be hypothetically removed
   *                           from its district
   * 
   * @return true iff the district will be severed
   */
   bool IsDistrictSevered(Node *proposed_node);
 
-  /*
+  /**
   * Queries whether or not a path exists between the start node and the target
   * node. Path is only valid if all nodes traversed in the path are in the
   * same district.
   *
-  * @param   start   The node to start the search at
-  * @param   target  The node to look for
+  * @param        start        the node to start the search at
+  * @param        target       the node to look for
   *
   * @return true iff a path exists between start and target and nodes traversed
   * belong in the same district
   */
   bool DoesPathExist(Node *start, Node *target);
 
+  /**
+   * Gets the list of changes of the graph on every step of the walk. The
+   * changes are represented as a vector of uint32_t, where each index of the
+   * vector represents the precinct ID, and the value at the index represents
+   * the new district ID of that precinct.
+   * 
+   * @return a vector of vector of uint32_t, representing changes at every step
+   */
+  vector<vector<uint32_t>&>& getMaps();
+
+  /**
+  * Gets the list of scores for every step of the walk. The scores consist of
+  * the total score, the compactness score, the population distribution score,
+  * the existing-border respectfulness score, and the VRA score. These are
+  * stored in a map from score name to score value.
+  * 
+  * @returns a vector of maps representing the scoring for each of the
+  *          maps generated so far in the walk.
+  */
+  vector<map<string, double> &>& getScores();
```

Comment:
Does this mean you're allocating these on the heap and cleaning them up later yourselves? If not, return a copy and not a reference.

Reply:
Yes, we are allocating it on the heap. We are destructing it in the Runner destructor.